python/lsst/pipe/tasks/imageDifference.py

```diff
@@ -518,7 +526,10 @@ def run(self, sensorRef, templateIdList=None):
 
             if self.config.doMeasurement:
                 self.log.info("Running diaSource measurement")
-                self.measurement.run(diaSources, subtractedExposure)
+                if not self.config.doMeasurementNew:
+                    self.measurement.run(diaSources, subtractedExposure)
+                else:
+                    self.measurement.run(diaSources, subtractedExposure, exposure, templateExposure)
```

Comment:
This suggests that running with different arguments will result in different fields being filled in (or even different schema).  Is it possible to have two different tasks, so we can simply delete one run call when we decide to remove the old way of doing things?